### PR TITLE
feat: setup a new next app on studio next

### LINKED ISSUE #727: Setup NextJS app
Related to #725

### PR DESCRIPTION
**Pull request by Shurtu-gal**
**Description**

- Setup a new next app in studio-next folder

**Related issue(s)**

Resolves #727

### CODE REVIEW

**Comment by fmvilas** [comment]
@Shurtu-gal linting is failing. Can you have a look?

**Comment by Shurtu-gal** [comment]
@fmvilas Done it!

**Review by fmvilas — APPROVED** [review]
LGTM 👍

File: apps/studio-next/.gitignore [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
+
+# dependencies
+/node_modules
```
Comment (KhudaDad414) [inline apps/studio-next/.gitignore:4]:
```suggestion
```

beign ignored in global .gitignore

File: apps/studio-next/src/app/page.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
+import Image from 'next/image'
+
+export default function Home() {
+  return (
+    <main className="flex min-h-screen flex-col items-center justify-between p-24">
```
Comment (KhudaDad414) [inline apps/studio-next/src/app/page.tsx:5]: what about clearing the template and only have a text saying `Home`? we are going to remove these anyway.

File: apps/studio-next/public/next.svg
Comment (KhudaDad414) [inline apps/studio-next/public/next.svg:1]: we can remove this these svgs I guess.

File: apps/studio-next/src/app/favicon.ico
Comment (KhudaDad414) [inline apps/studio-next/src/app/favicon.ico:1]: what about using [studio's favicon.ico] here?

**Comment by Shurtu-gal** [comment]
Sure, I will make the changes.

**Comment by Shurtu-gal** [comment]
@KhudaDad414 All changes are done.

**Review by KhudaDad414 — DISMISSED** [review]

**Comment by Amzani** [comment]
@KhudaDad414 @smoya do you have access to Netlify of studio ? we need add  a new domain `https://studio-next.asyncapi.com/` that point to right publish directory (cc @Shurtu-gal )

**Comment by KhudaDad414** [comment]
@Amzani, @derberg might be able to help here.

**Comment by smoya** [comment]
> @KhudaDad414 @smoya do you have access to Netlify of studio ? we need add  a new domain `https://studio-next.asyncapi.com/` that point to right publish directory (cc @Shurtu-gal )

Not me. Btw, not sure if Netlify manages the DNS for Studio. I think its all in a domain provider and most probably managed by Linux Foundation. Anyway, @derberg @fmvilas can answer.

File: package.json [hunk 3]
```diff
@@ -16,7 +16,8 @@
     "version-packages": "changeset version",
     "publish": "changeset publish",
     "build:studio": "turbo run build --no-cache --filter=studio...",
-    "build:ds": "turbo run build --filter=design-system..."
+    "build:ds": "turbo run build --filter=design-system...",
+    "build:next": "turbo run build --no-cache --filter=studio-next..."
```
Comment (Amzani) [inline package.json:20]: @Shurtu-gal I suggest `build:studio-next`

Looks like you have installed `next` as a global dependency ?  When I run `npm run build:next` I have this error

```
npm run build:next

> build:next
> turbo run build --no-cache --filter=studio-next...

╭────────────────────────────────────────────────────────────────────────╮
│                                                                        │
│                  Update available v1.10.7 ≫ v1.10.12                   │
│    Changelog: https://github.com/vercel/turbo/releases/tag/v1.10.12    │
│               Run "npx @turbo/codemod update" to update                │
│                                                                        │
│      Follow @turborepo for updates: https://twitter.com/turborepo      │
╰────────────────────────────────────────────────────────────────────────╯
• Packages in scope: studio-next
• Running build in 1 packages
• Remote caching disabled
studio-next:build: cache bypass, force executing baa9d44d992a27f4
studio-next:build:
studio-next:build: > studio-next@0.1.0 build
studio-next:build: > next build
studio-next:build:
studio-next:build: sh: next: command not found
studio-next:build: npm ERR! Lifecycle script `build` failed with error:
studio-next:build: npm ERR! Error: command failed
studio-next:build: npm ERR!   in workspace: studio-next@0.1.0
studio-next:build: npm ERR!   at location: /Users/samir.amzani/Projects/asyncapi/studio/apps/studio-next
studio-next:build: ERROR: command finished with error: command (/Users/samir.amzani/Projects/asyncapi/studio/apps/studio-next) npm run build exited (1)
command (/Users/samir.amzani/Projects/asyncapi/studio/apps/studio-next) npm run build exited (1)

 Tasks:    0 successful, 1 total
Cached:    0 cached, 1 total
  Time:    1.038s
Failed:    studio-next#build

 ERROR  run failed: command  exited (1)
```

Comment (Amzani) [reply package.json:20]: When I try to install something in the new app 
```
npm install xxx --workspace=react-next

npm ERR! No workspaces found:
npm ERR!   --workspace=react-next

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/samir.amzani/.npm/_logs/2023-08-03T11_54_00_009Z-debug-0.log
➜  studio git:(8a12b9c) npm install nextjs --workspace=react-next
npm ERR! No workspaces found:
npm ERR!   --workspace=react-next

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/samir.amzani/.npm/_logs/2023-08-03T11_54_14_206Z-debug-0.log
```
I saw this happening in an other project, solved by creating a empty workspace: see [this](https://turbo.build/repo/docs/core-concepts/monorepos/code-generation#add-an-empty-workspace)

**Review by Amzani — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (Shurtu-gal) [reply package.json:20]: Let me correct it.

**Review by KhudaDad414 — APPROVED** [review]
LGTM 👍

**Comment by Shurtu-gal** [comment]
Hey, @KhudaDad414 Can this be merged as I have to work on further things #762?

**Review by fmvilas — APPROVED** [review]

**Pull request merged**